src/tools/chip-cert/Cmd_GenCert.cpp

```diff
@@ -1176,7 +1191,7 @@ bool Cmd_GenCert(int argc, char * argv[])
                    gFutureExtensionsCount, newCert.get(), newKey.get(), gCertConfig);
     VerifyTrueOrExit(res);
 
-    if (gCertConfig.IsErrorTestCaseEnabled() && IsChipCertFormat(gOutCertFormat))
+    if (IsChipCertFormat(gOutCertFormat) && (gCertConfig.IsErrorTestCaseEnabled() || gCertType == CertType::kNetworkIdentity))
```

Comment:
This could use a comment explaining why the network identity case is being treated like the "error test case enabled" case.